criu/proc_parse.c

```diff
@@ -651,7 +662,8 @@ static int handle_vma(pid_t pid, struct vma_area *vma_area, const char *file_pat
 
 				close_safe(vm_file_fd);
 				return 0;
-			}
+			} else /* for memfd_secret case */
```

Comment:
Are we certain that previous cases cover all possibilities? And even if they do, for future compatibility I think it should be
`else if (is_memfd_secret())` here

Reply:
I have thought of this before and I agree with you Mike. But, the issue is, call to `is_memfd_secret(st_buf->st_dev)` here always returns false. I'm not sure why.

Furthermore, the dump works even without this else block. But, the problem with this approach is `vma_area->e->status` is to `VMA_FILE_PRIVATE` not `VMA_AREA_MEMFD_SECRET` which is what we want.

Moreover, we are already in a parent `else` [block](https://github.com/checkpoint-restore/criu/blob/8ee13b6536cb441c4c515b32dec357a6fca9affb/criu/proc_parse.c#L642) without any condition, and currently at the end of it. So, it kind of make sense. Plus, there is already another if [block](https://github.com/checkpoint-restore/criu/blob/8ee13b6536cb441c4c515b32dec357a6fca9affb/criu/proc_parse.c#L668) to handle other cases if needed.

Reply:
done!